src/chi/components/buttons/buttons.scss

```diff
@@ -539,10 +546,240 @@ $sizes: (
             box-shadow: 0 0 0 3px rgba(set-color($color), 0.4);
 
             .a-icon {
+              color: $text-color;
+            }
+          }
+
+          &:active,
+          &.-active {
+            &,
+            &:focus,
+            &.-focus {
+              box-shadow: none;
+            }
+          }
+        }
+
+        &.-flat {
+          background-color: transparent;
+          border: $border solid transparent;
+          box-shadow: none;
+          color: set-color($color);
+
+          &:hover:not(.-icon),
+          &.-hover:not(.-icon) {
+            background-color: set-color(transparent);
+            border: $border solid set-color($color);
+            color: set-color($color);
+          }
+
+          &:focus,
+          &.-focus {
+            background-color: set-color(transparent);
+            border: $border solid set-color($color);
+            box-shadow: 0 0 0 3px rgba(set-color($color, $tone-focus-shadow), 0.4);
+          }
+
+          &:active,
+          &.-active {
+            &,
+            &:focus,
+            &.-focus {
+              background-color: transparent;
+              border: $border solid set-color($color);
+              box-shadow: none;
               color: set-color($color);
             }
           }
 
+          .a-icon {
+            color: $text-white;
+          }
+
+          &.-icon {
+            & .a-icon {
+              color: set-color(grey, 50);
+            }
+
+            &:hover,
+            &.-hover {
+              background-color: transparent;
+              border: $border solid transparent;
```

Comment:
There's a bug with the focus state. Border is getting overridden on :hover:focus with this transparent border color value.